exchanges/orderbook/tranches_test.go

```diff
@@ -9,380 +9,249 @@ import (
-	Item{Price: 1333, Amount: 1},
-	Item{Price: 1332, Amount: 1},
-	Item{Price: 1331, Amount: 1},
-	Item{Price: 1330, Amount: 1},
-	Item{Price: 1329, Amount: 1},
-	Item{Price: 1328, Amount: 1},
-	Item{Price: 1327, Amount: 1},
-	Item{Price: 1326, Amount: 1},
-	Item{Price: 1325, Amount: 1},
-	Item{Price: 1324, Amount: 1},
-	Item{Price: 1323, Amount: 1},
-	Item{Price: 1322, Amount: 1},
-	Item{Price: 1321, Amount: 1},
-	Item{Price: 1320, Amount: 1},
-	Item{Price: 1319, Amount: 1},
-	Item{Price: 1318, Amount: 1},
-	Item{Price: 1317, Amount: 1},
+var bid = Tranches{
+	{Price: 1336, Amount: 1},
+	{Price: 1335, Amount: 1},
+	{Price: 1334, Amount: 1},
+	{Price: 1333, Amount: 1},
+	{Price: 1332, Amount: 1},
+	{Price: 1331, Amount: 1},
+	{Price: 1330, Amount: 1},
+	{Price: 1329, Amount: 1},
+	{Price: 1328, Amount: 1},
+	{Price: 1327, Amount: 1},
+	{Price: 1326, Amount: 1},
+	{Price: 1325, Amount: 1},
+	{Price: 1324, Amount: 1},
+	{Price: 1323, Amount: 1},
+	{Price: 1322, Amount: 1},
+	{Price: 1321, Amount: 1},
+	{Price: 1320, Amount: 1},
+	{Price: 1319, Amount: 1},
+	{Price: 1318, Amount: 1},
+	{Price: 1317, Amount: 1},
 }
 
 // Display displays depth content for tests
-func (ll *linkedList) display() {
-	for tip := ll.head; tip != nil; tip = tip.Next {
-		fmt.Printf("NODE: %+v %p \n", tip, tip)
+func (ts Tranches) display() {
+	for x := range ts {
+		fmt.Printf("Tranche: %+v %p \n", ts[x], &ts[x])
 	}
 	fmt.Println()
 }
 
 func TestLoad(t *testing.T) {
-	list := asks{}
+	list := askTranches{}
 	Check(t, list, 0, 0, 0)
 
-	stack := newStack()
-	list.load(Items{
+	list.load(Tranches{
 		{Price: 1, Amount: 1},
 		{Price: 3, Amount: 1},
 		{Price: 5, Amount: 1},
 		{Price: 7, Amount: 1},
 		{Price: 9, Amount: 1},
 		{Price: 11, Amount: 1},
-	}, stack, time.Now())
-
-	if stack.getCount() != 0 {
-		t.Fatalf("incorrect stack count expected: %v received: %v", 0, stack.getCount())
-	}
+	})
 
 	Check(t, list, 6, 36, 6)
 
-	list.load(Items{
+	list.load(Tranches{
 		{Price: 1, Amount: 1},
 		{Price: 3, Amount: 1},
 		{Price: 5, Amount: 1},
-	}, stack, time.Now())
-
-	if stack.getCount() != 3 {
-		t.Fatalf("incorrect stack count expected: %v received: %v", 3, stack.getCount())
-	}
+	})
 
 	Check(t, list, 3, 9, 3)
 
-	list.load(Items{
+	list.load(Tranches{
 		{Price: 1, Amount: 1},
 		{Price: 3, Amount: 1},
 		{Price: 5, Amount: 1},
 		{Price: 7, Amount: 1},
-	}, stack, time.Now())
-
-	if stack.getCount() != 2 {
-		t.Fatalf("incorrect stack count expected: %v received: %v", 2, stack.getCount())
-	}
+	})
 
 	Check(t, list, 4, 16, 4)
 
 	// purge entire list
-	list.load(nil, stack, time.Now())
-
-	if stack.getCount() != 6 {
-		t.Fatalf("incorrect stack count expected: %v received: %v", 6, stack.getCount())
-	}
-
+	list.load(nil)
 	Check(t, list, 0, 0, 0)
 }
 
-// 22222386	        57.3 ns/op	       0 B/op	       0 allocs/op (old)
-// 27906781	        42.4 ns/op	       0 B/op	       0 allocs/op (new)
+// 27906781	        42.4 ns/op	       0 B/op	       0 allocs/op (old)
```

Comment:
master:

```
18274713	        56.66 ns/op	       0 B/op	       0 allocs/op
```

This branch:

```
87015793	        14.03 ns/op	       0 B/op	       0 allocs/op
```

It seems quite similar, as for the report allocs, I usually don't include them because it seems to get displayed in vscode as default. If you need me to add them I will.